sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/kafka/AzureEventHubsKafkaOAUTH2Configuration.java

```diff
@@ -0,0 +1,108 @@
+import com.azure.spring.cloud.autoconfigure.context.AzureTokenCredentialAutoConfiguration;
+import com.azure.spring.cloud.core.implementation.credential.resolver.AzureTokenCredentialResolver;
+import com.azure.spring.cloud.core.implementation.properties.AzureThirdPartyServiceProperties;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.kafka.DefaultKafkaConsumerFactoryCustomizer;
+import org.springframework.boot.autoconfigure.kafka.DefaultKafkaProducerFactoryCustomizer;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder;
+import org.springframework.cloud.stream.binder.kafka.config.KafkaBinderConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.kafka.core.KafkaTemplate;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.azure.spring.cloud.autoconfigure.context.AzureContextUtils.DEFAULT_TOKEN_CREDENTIAL_BEAN_NAME;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaAutoconfigurationUtils.KAFKA_OAUTH_CONFIGS;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaAutoconfigurationUtils.buildAzureProperties;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaAutoconfigurationUtils.needConfigureSaslOAuth;
+import static com.azure.spring.cloud.core.implementation.util.AzureConfigUtils.AZURE_TOKEN_CREDENTIAL;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for Azure Event Hubs Kafka support. Provide Managed Identity-based
+ * OAuth2 authentication for Event Hubs for Kafka on the basis of Spring Boot Autoconfiguration.
+ *
+ * @since 4.3.0
+ */
+@Configuration(proxyBeanMethods = false)
+@Import(AzureEventHubsKafkaOAUTH2Configuration.AzureKafkaSpringCloudStreamAutoConfiguration.class)
+@ConditionalOnClass(KafkaTemplate.class)
+@AutoConfigureAfter(AzureTokenCredentialAutoConfiguration.class)
+public class AzureEventHubsKafkaOAUTH2Configuration {
+
+    private final KafkaProperties kafkaProperties;
+    private final AzureTokenCredentialResolver tokenCredentialResolver;
+    private final AzureGlobalProperties azureGlobalProperties;
+    private final TokenCredential defaultTokenCredential;
+
+    private static final String SPRING_BOOT_KAFKA_PROPERTIES_BEAN_NAME = "spring.kafka-org.springframework.boot"
+        + ".autoconfigure.kafka.KafkaProperties";
+
+    AzureEventHubsKafkaOAUTH2Configuration(@Qualifier(SPRING_BOOT_KAFKA_PROPERTIES_BEAN_NAME) KafkaProperties kafkaProperties,
```

Comment:
This autoconfiguration needs to live together with our legacy kafka autoconfiguration, thus there will be two KafkaProperties in some cases. 
Do we consider to add other conditions of this autoconfiguration, like condition on no connection-string providers for event hubs since this configuration is for credential-free cases